src/Controls/tests/TestCases.HostApp/Issues/Issue23424.xaml.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace Maui.Controls.Sample.Issues;
+
+[XamlCompilation(XamlCompilationOptions.Compile)]
+[Issue(IssueTracker.Github, "23424", "BackButtonBehavior IsVisible=False does not hide the back button", PlatformAffected.All)]
+public partial class Issue23424 : Shell
+{
+	public Issue23424()
+	{
+		Items.Add(new ContentPage());
+		Routing.RegisterRoute(nameof(DetailPage),typeof(DetailPage));
+		GoToAsync(nameof(DetailPage));
```

Comment:
I don't think we need to register a route here. 

Maybe just this to keep it simple
```C#
Items.Add(new ContentPage());
		Navigation.PushAsync(new DetailPage());
```